### PR TITLE
SMP: Documented some functions used in the Orbifold Tutte parameterizer

### PR DESCRIPTION
**Pull request by MaelRL**
## Summary of Changes

The `orbifold.cpp` example relied on internal functions. Some are removed, some are documented. When they are documented, their API is improved.

Partial fix for https://github.com/CGAL/cgal/issues/2123#issuecomment-358584683.

## Release Management

* Affected package(s): Surface Mesh Parameterization
* Issue(s) solved (if any): --
* Feature/Small Feature (if any): [SMP Improve Orbifold Tutte documentation](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/SMP_Improve_Orbifold_Tutte_documentation)
* Pre-approved by @sloriot  12 February 2018

### CODE REVIEW

**Comment by lrineau** [comment]
Is there a small feature where the documentation changes are published and reviewed?

**Comment by MaelRL** [comment]
There is now.

Unrelated: Is there a reason why the container cpp concepts (`Container`, `SequenceContainer`, etc.) are not in `Documentation/General.txt` but the iterator concepts (`ForwardIterator`, etc.) are?

File: Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h [hunk 1]
```diff
@@ -64,20 +66,276 @@ namespace CGAL {
...
+/// \ingroup PkgSurfaceParameterizationOrbifoldHelperFunctions
 ///
-/// Weight type used in the parameterization computation.
+/// Read a serie of cones from an input stream. Cones are passed as an
+/// integer value that is the index of a vertex handle in the mesh `pm`, using
+/// the vertex index property map `vpmap` for correspondency.
 ///
-/// MVC weights are guaranteed to generate positive edge weights, and the parameterization
-/// is guaranteed to be injective.
+/// \attention The mesh is here `sm`, it is the base mesh of the CGAL::Seam_mesh
```
Comment (lrineau) [inline Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h]: You are talking about `tm`, not `sm`.

File: Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h [hunk 2]
```diff
@@ -64,20 +66,276 @@ namespace CGAL {
...
+Error_code read_cones(const TriangleMesh& tm, const char* filename, ConeOutputIterator out)
+{
+  std::ifstream in(filename);
+  return read_cones(tm, in, out);
+}
+
+/// \ingroup PkgSurfaceParameterizationOrbifoldHelperFunctions
+///
+/// Locate the cones on the seam mesh (that is, find the corresponding seam mesh
+/// vertex_descriptor) and mark them with a tag to indicate whether the cone is a
```
Comment (sloriot) [inline Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h]: backticks

File: Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h [hunk 3]
```diff
@@ -64,20 +66,276 @@ namespace CGAL {
...
+/// simple cone or a duplicated cone (see \link PkgSurfaceParameterizationEnums Cone_type \endlink).
+///
+/// \attention The cones must be ordered: the first and last cones are the extremetities of the seam.
+///
+/// \tparam SeamMesh is the same class CGAL::Seam_mesh that is passed to the parameterized.
```
Comment (sloriot) [inline Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h]: backticks

File: Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h [hunk 4]
```diff
@@ -64,20 +66,276 @@ namespace CGAL {
...
+///                  It is a template parameter for convenience (not having to rewrite the template
+///                  parameters of the class CGAL::Seam_mesh).
```
Comment (sloriot) [inline Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h]: backticks

File: Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h [hunk 5]
```diff
@@ -64,20 +66,276 @@ namespace CGAL {
...
+/// \tparam ConeInputBidirectionalIterator must be a model of `BidirectionalIterator`
+///                  with value type `boost::graph_traits<SeamMesh::TriangleMesh>::%vertex_descriptor`.
+/// \tparam ConeMap must be a model of <a href="http://en.cppreference.com/w/cpp/concept/AssociativeContainer"><tt>AssociativeContainer</tt></a>
+///                 with `boost::graph_traits<SeamMesh>::%vertex_descriptor` as key type and
+///                 \link PkgSurfaceParameterizationEnums Cone_type \endlink as value type.
+///
+/// \param mesh the seam mesh
+/// \param first, beyond the range of cones, as vertex descriptors of the base mesh.
+/// \param cones a writable property map that will store the cones as vertex descriptors
```
Comment (sloriot) [inline Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/Orbifold_Tutte_parameterizer_3.h]: associative container or writable property map?

File: Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/orbifold_shortest_path.h [hunk 6]
```diff
@@ -97,11 +101,29 @@ class Stop_at_target_Dijkstra_visitor : boost::default_dijkstra_visitor
   }
 };
 
-template<typename TriangleMesh, typename OutputIterator>
+} // namespace internal
+
+/// \ingroup PkgSurfaceParameterizationOrbifoldHelperFunctions
+///
+/// Compute the shortest path between `source` and `target` over `mesh`, using
+/// <a href="http://www.boost.org/doc/libs/1_55_0/libs/graph/doc/dijkstra_shortest_paths.html">
```
Comment (sloriot) [inline Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/orbifold_shortest_path.h]: use a url without version

**Comment by lrineau** [comment]
Was successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-173

But only pre-approved two days ago.

**Comment by afabri** [comment]
It is unrealistic to assume that another editor will have a look and cry in the next 12 days.

**Comment by lrineau** [comment]
That was now one week ago. I will merge it today.

**Comment by lrineau** [comment]
Successfully tested in [CGAL-4.12-Ic-176], and then merged.

[CGAL-4.12-Ic-176]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-176.shtml

**Pull request merged**